### PR TITLE
Change CSV to XLS committee members export

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Controller/EnMarche/CommitteeManagerController.php [hunk 1]
```diff
@@ -117,9 +119,26 @@ public function exportMembersAction(Request $request, Committee $committee): Res
         $uuids = GroupUtils::getUuidsFromJson($request->request->get('exports', ''));
         $adherents = GroupUtils::removeUnknownAdherents($uuids, $committeeManager->getCommitteeMembers($committee));
 
-        return new Response(AdherentCsvSerializer::serialize($adherents ?? []), 200, [
-            'Content-Type' => 'text/csv',
-            'Content-Disposition' => 'attachment; filename="membres-du-comite.csv"',
+        return new Response(
+            $serializer->serialize(
+                $adherents,
+                'xlsx',
```
Comment (ottaviano) [inline src/Controller/EnMarche/CommitteeManagerController.php]: `XlsxEncoder::FORMAT`

File: src/Controller/EnMarche/CommitteeManagerController.php [hunk 2]
```diff
@@ -117,9 +119,26 @@ public function exportMembersAction(Request $request, Committee $committee): Res
...
+                [
+                    'groups' => ['export'],
+                    DateTimeNormalizer::FORMAT_KEY => 'Y-m-d',
+                    'headers' => [
```
Comment (ottaviano) [inline src/Controller/EnMarche/CommitteeManagerController.php]: `export_file_headers`?

File: src/Serializer/XlsxEncoder.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        return $value;
+    }
+
+    public function supportsEncoding($format)
+    {
+        return self::FORMAT === $format;
+    }
+
+    public function supportsDecoding($format)
```
Comment (ottaviano) [inline src/Serializer/XlsxEncoder.php]: why do you implement this method too?

File: src/Serializer/XlsxEncoder.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+use Symfony\Component\Serializer\Encoder\EncoderInterface;
+
+class XlsxEncoder implements EncoderInterface
+{
+    public const FORMAT = 'xlsx';
+
+    public function encode($data, $format, array $context = [])
+    {
+        if (!is_iterable($data)) {
+            throw new \InvalidArgumentException('This method requires a collection of entities');
```
Comment (ottaviano) [inline src/Serializer/XlsxEncoder.php]: normally it cannot be entities here, as we are in encoding process 
`This method requires an array of data`?

File: src/Serializer/XlsxEncoder.php (hunk 3 above)
Comment (illusionOfParadise) [reply src/Serializer/XlsxEncoder.php]: Good question!

**Comment by thomasbisignani** [comment]
Nice encoder :ok_hand:

File: src/Controller/EnMarche/CommitteeManagerController.php [hunk 5]
```diff
@@ -117,9 +122,26 @@ public function exportMembersAction(Request $request, Committee $committee): Res
...
-        return new Response(AdherentCsvSerializer::serialize($adherents ?? []), 200, [
-            'Content-Type' => 'text/csv',
-            'Content-Disposition' => 'attachment; filename="membres-du-comite.csv"',
+        return new Response(
+            $serializer->serialize(
+                $adherents,
+                XlsxEncoder::FORMAT,
+                [
+                    'groups' => ['export'],
+                    DateTimeNormalizer::FORMAT_KEY => 'Y-m-d',
```
Comment (ottaviano) [inline src/Controller/EnMarche/CommitteeManagerController.php]: what about using french format: `d/m/Y`? more user-friendly :)

**Pull request merged**